### PR TITLE
feat: setup logging with tracing

### PR DESCRIPTION
**Pull request by FMFigueroa**

### CODE REVIEW

File: crates/http-server/src/cli.rs [hunk 1]
```diff
@@ -9,12 +9,12 @@ pub struct Cli {
     #[arg(long, default_value = "127.0.0.1")]
     pub host: IpAddr,
     /// The port to listen on
-    #[arg(long, default_value = "7878")]
+    #[arg(long, default_value = "3000")]
```
Comment (LeoBorai) [inline crates/http-server/src/cli.rs]: Can we keep this as `7878`? Which is the one used in [The Rust Programming Language Book](https://doc.rust-lang.org/book/ch20-01-single-threaded.html#listening-to-the-tcp-connection)?

> Using TcpListener, we can listen for TCP connections at the address 127.0.0.1:7878. In the address, the section before the colon is an IP address representing your computer (this is the same on every computer and doesn’t represent the authors’ computer specifically), and 7878 is the port. We’ve chosen this port for two reasons: HTTP isn’t normally accepted on this port so our server is unlikely to conflict with any other web server you might have running on your machine, and 7878 is rust typed on a telephone.

File: crates/http-server/src/cli.rs [hunk 2]
```diff
@@ -9,12 +9,12 @@ pub struct Cli {
...
     pub port: u16,
 }
 
 impl Cli {
     pub fn address(&self) -> SocketAddr {
         SocketAddr::new(self.host, self.port)
     }
-}
+}
```
Comment (LeoBorai) [inline crates/http-server/src/cli.rs]: Please keep the trailing newline

File: crates/http-server/src/main.rs [hunk 3]
```diff
@@ -22,4 +23,4 @@ async fn main() -> Result<()> {
         .await?;
 
     Ok(())
-}
+}
```
Comment (LeoBorai) [inline crates/http-server/src/main.rs]: Please keep the trailing newline

File: crates/http-server/src/server/mod.rs [hunk 4]
```diff
@@ -1,25 +1,45 @@
...
+
+async fn handler_default() -> impl IntoResponse{
+   debug!("A request has been received on the default route");
+   Html("Hello <strong>Rust!!!</strong>")
+}
+
+async fn handler_custom() -> impl IntoResponse {
+    info!("A request has been received on a custom route");
+    Html("Hello <strong>Rust!!!</strong>")
+}
```
Comment (LeoBorai) [inline crates/http-server/src/server/mod.rs]: Please keep the trailing newline

File: crates/http-server/src/server/mod.rs [hunk 5]
```diff
@@ -1,25 +1,45 @@
...
 }
+
+async fn handler_default() -> impl IntoResponse{
+   debug!("A request has been received on the default route");
+   Html("Hello <strong>Rust!!!</strong>")
+}
+
+async fn handler_custom() -> impl IntoResponse {
+    info!("A request has been received on a custom route");
+    Html("Hello <strong>Rust!!!</strong>")
```
Comment (LeoBorai) [inline crates/http-server/src/server/mod.rs]: Can we remove this as not related to tracing?

**Review by LeoBorai — COMMENTED** [review]
Nice work setting up tracing!
Lets clean up a bit test code and take back [trailing newlines please](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline#:~:text=The%20way%20Unix%20regards%20its,an%20%5Cn%20to%20end%20it.).

**Comment by FMFigueroa** [comment]
It's something very simple... it could be improved later...
at least it's well formatted !

**Comment by LeoBorai** [comment]
Thanks @FMFigueroa! Can you run `cargo fmt` please?

File: crates/http-server/src/server/mod.rs [hunk 6]
```diff
@@ -1,8 +1,9 @@
-use axum::Router;
-
 use crate::cli::Cli;
+use axum::{routing::get, Router};
+use tracing::{info, Level};
+use tracing_subscriber::EnvFilter;
 
-pub struct Server(Router);
+pub struct Server(pub Router);
```
Comment (LeoBorai) [inline crates/http-server/src/server/mod.rs:6]: Lets keep the `Router` encapsulated and use the getter we have as a method instead

File: crates/http-server/src/server/mod.rs [hunk 7]
```diff
@@ -12,14 +13,18 @@ impl Server {
 
 impl From<Cli> for Server {
     fn from(value: Cli) -> Self {
+        let filter = EnvFilter::from_default_env().add_directive(Level::INFO.into());
+
+        tracing_subscriber::fmt().with_env_filter(filter).init();
```
Comment (LeoBorai) [inline crates/http-server/src/server/mod.rs:18]: Taking advantage of the `Cli` I think we could have a `log` argument we could use to pass it to the `tracing_subscriber`. This way we could do:

```bash
http-server --log debug
```

This would set the `Level::DEBUG` in L16.

**Comment by FMFigueroa** [comment]
> Thanks @FMFigueroa! Can you run `cargo fmt` please?

Sure.. Apology man..!

File: crates/http-server/src/cli.rs [hunk 8]
```diff
@@ -11,10 +14,16 @@ pub struct Cli {
     /// The port to listen on
     #[arg(long, default_value = "7878")]
     pub port: u16,
+    /// Directory to serve files from
+    #[arg(long, default_value = "./")]
+    pub root_dir: PathBuf,
```
Comment (LeoBorai) [inline crates/http-server/src/cli.rs]: Can we move this into its own PR? Given that is out of scope of tracing?

**Review by LeoBorai — COMMENTED** [review]
Lets merge tracing first and then move into file explorer pls

**Review by LeoBorai — APPROVED** [review]
LGTM!
